### PR TITLE
Bounding Box for Selection

### LINKED ISSUE #468: Dynamic Selection Colors Based on Object Color
Currently, the selection color is static and defaults to blue. While this works, it may not always provide the best visual clarity, especially when working with a variety of object colors. The static blue selection color can sometimes clash with the colors of certain objects, making it difficult to distinguish selections and reducing overall visual appeal.

We suggest making the selection color dynamic, where the selection highlight color is based on the object's original color. 
A slight alteration in brightness or saturation could be used to highlight the selected object without drastically changing its appearance.

### PR DESCRIPTION
**Pull request by arjxn-py**
https://github.com/user-attachments/assets/887b59da-aaf9-4265-b8cc-6c814ebc7d90

Shall fix #468

### CODE REVIEW

File: packages/base/src/3dview/mainview.tsx [hunk 1]
```diff
@@ -555,8 +555,12 @@ export class MainView extends React.Component<IProps, IStates> {
     if (intersects.length > 0) {
       // Find the first intersection with a visible object
       for (const intersect of intersects as ILineIntersection[]) {
-        // Object is hidden
-        if (!intersect.object.visible || !intersect.object.parent?.visible) {
+        // Object is hidden or a bounding box
+        if (
+          !intersect.object.visible ||
+          !intersect.object.parent?.visible ||
+          intersect.object.name === 'selectionBoundingBox'
```
Comment (trungleduc) [inline packages/base/src/3dview/mainview.tsx:562]: Let's make `selectionBoundingBox` a constant variable

**Comment by trungleduc** [comment]
holding CTRL and click on the object one more time creates a much bigger bounding box. 


https://github.com/user-attachments/assets/e6150fbb-2924-44aa-8383-73b2973f1042

**Comment by trungleduc** [comment]
yeah, tested with the lasted preview link and it works well

**Comment by trungleduc** [comment]
Thanks!

**Comment by arjxn-py** [comment]
There's one more thing wrong related to the color update when changing color. Looks like I've removed something that wasn't meant to be.


https://github.com/user-attachments/assets/784b8a54-bee5-4c95-b33e-4ad9975c5120

**Comment by arjxn-py** [comment]
> There's one more thing wrong related to the color update when changing color. Looks like I've removed something that wasn't meant to be.

Fixed.


https://github.com/user-attachments/assets/f7831e22-b4c9-4833-8662-4677400daa4e

**Comment by arjxn-py** [comment]
Bot please update snapshots

**Review by martinRenou — CHANGES_REQUESTED** [review]
I still see a tiny issue, it seems there is still a place where we are setting the selected mesh color to `SELECTED_MESH_COLOR`. Because when I change the color to red, it's actually going to `SELECTED_MESH_COLOR` first.

Actually I would suggest getting rid of the `SELECTED_MESH_COLOR` as we won't be using it anymore. You could define your own constant `BOUNDING_BOXES_COLOR`.


https://github.com/user-attachments/assets/9421ffa8-2ded-425d-a29d-73f9a7a3c635

**Comment by arjxn-py** [comment]
Done. Thanks @martinRenou


https://github.com/user-attachments/assets/97b4e025-789d-47b0-89d0-8c1dc01ad123

**Review by martinRenou — APPROVED** [review]
Thanks!

**Pull request merged**